### PR TITLE
Integrate abt + fiam

### PR DESCRIPTION
**Pull request by JasonAHeron**

### CODE REVIEW

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/AbtIntegrationHelper.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/** @hide */
+public class AbtIntegrationHelper {
+  private static FirebaseABTesting abTesting;
+
+  @Inject
+  public AbtIntegrationHelper(FirebaseABTesting abTesting) {
+    this.abTesting = abTesting;
+  }
+
+  public void updateRunningExperiments(FetchEligibleCampaignsResponse response) {
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/AbtIntegrationHelper.java]: Lets add some docs on what this method does

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java [hunk 2]
```diff
@@ -106,9 +108,19 @@ private static boolean hasAnalyticsTrigger(TriggeringCondition tc, String event)
...
+    long campaignEndTime;
+    if (content.getPayloadCase().equals(ThickContent.PayloadCase.VANILLA_PAYLOAD)) {
+      campaignStartTime = content.getVanillaPayload().getCampaignStartTimeMillis();
+      campaignEndTime = content.getVanillaPayload().getCampaignEndTimeMillis();
+    } else {
+      campaignStartTime =
+          content.getExperimentalPayload().getExperimentPayload().getExperimentStartTimeMillis();
+      campaignEndTime =
+          campaignStartTime
+              + content.getExperimentalPayload().getExperimentPayload().getTimeToLiveMillis();
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java]: Have we validated that this actually mimics what the user sets in the console for the expt?

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/injection/modules/AbTestingModule.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+package com.google.firebase.inappmessaging.internal.injection.modules;
+
+import com.google.firebase.abt.FirebaseABTesting;
+import com.google.firebase.inappmessaging.internal.AbtIntegrationHelper;
+import dagger.Module;
+import dagger.Provides;
+
+/**
+ * Provider for ABTesting
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/injection/modules/AbTestingModule.java]: Module

File: firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java [hunk 4]
```diff
@@ -259,15 +285,43 @@ public void stream_onExpiredCampaign_doesNotTrigger() {
...
-    String unrelatedAnalyticsEvent = "some_other_event";
-    when(mockApiClient.getFiams(CAMPAIGN_IMPRESSIONS)).thenReturn(campaignsResponse);
+  public void stream_onExpiredExperiment_doesNotTrigger() {
+    ThickContent t =
+        ThickContent.newBuilder(thickContent)
+            .clearContent()
+            .setExperimentalPayload(
+                CampaignProto.ExperimentalCampaignPayload.newBuilder()
+                    .setExperimentPayload(
+                        FirebaseAbt.ExperimentPayload.newBuilder()
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java]: bug?

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java (hunk 2 above)
Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java]: Right now this does not, in the future we need to make a backend change to sync this. b/147681213

File: firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java (hunk 4 above)
Comment (JasonAHeron) [reply firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java]: fixed!

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/injection/modules/AbTestingModule.java (hunk 3 above)
Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/injection/modules/AbTestingModule.java]: done

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/AbtIntegrationHelper.java (hunk 1 above)
Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/AbtIntegrationHelper.java]: done

**Pull request merged**